packages/ui/src/utils/setUserAgent/constants.ts

```diff
@@ -1,24 +1,80 @@
 import {
-  setCustomUserAgent,
-  SetCustomUserAgentInput,
+  AuthUserAgentInput,
+  GeoUserAgentInput,
+  InAppMessagingUserAgentInput,
+  StorageUserAgentInput,
   AuthAction,
+  GeoAction,
+  InAppMessagingAction,
+  StorageAction,
   Category,
 } from '@aws-amplify/core/internals/utils';
 
-// replace type once Category input types are available
-export const ACCOUNT_SETTINGS_INPUT_BASE: Omit<
-  SetCustomUserAgentInput,
+export const AUTHENTICATOR_INPUT_BASE: Omit<
+  AuthUserAgentInput,
   'additionalDetails'
 > = {
-  apis: [AuthAction.DeleteUser, AuthAction.UpdatePassword],
+  apis: [
+    AuthAction.SignUp,
+    AuthAction.ConfirmSignUp,
+    AuthAction.ResendSignUpCode,
+    AuthAction.SignIn,
+    AuthAction.ConfirmSignIn,
+    AuthAction.FetchUserAttributes,
+    AuthAction.SignOut,
+    AuthAction.ResetPassword,
+    AuthAction.ConfirmResetPassword,
+    AuthAction.SignInWithRedirect,
+  ],
+  category: Category.Auth,
+};
+
+export const CHANGE_PASSWORD_INPUT_BASE: Omit<
+  AuthUserAgentInput,
+  'additionalDetails'
+> = {
+  apis: [AuthAction.UpdatePassword],
+  category: Category.Auth,
+};
+
+export const DELETE_USER_INPUT_BASE: Omit<
+  AuthUserAgentInput,
+  'additionalDetails'
+> = {
+  apis: [AuthAction.DeleteUser],
   category: Category.Auth,
 };
 
-// replace type once Category input types are available
 export const IN_APP_MESSAGING_INPUT_BASE: Omit<
-  SetCustomUserAgentInput,
+  InAppMessagingUserAgentInput,
   'additionalDetails'
 > = {
-  apis: undefined, // TODO
+  apis: [InAppMessagingAction.DispatchEvent],
   category: Category.InAppMessaging,
 };
+
+export const LOCATION_SEARCH_INPUT_BASE: Omit<
+  GeoUserAgentInput,
+  'additionalDetails'
+> = {
+  category: Category.Geo,
+  apis: [
+    GeoAction.SearchByText,
+    GeoAction.SearchForSuggestions,
+    GeoAction.SearchByPlaceId,
+  ],
+};
+
+export const MAP_VIEW_INPUT_BASE: Omit<GeoUserAgentInput, 'additionalDetails'> =
+  {
+    category: Category.Geo,
+    apis: [],
```

Comment:
Does `MapView` not have any network call API interactions?

Reply:
correct it does not. Thad said it just uses map name to create a request to location service

Reply:
@thaddmt Can you confirm whether the `MapView` any Geo JS APIs should be included here? If not think `MapView` can be omitted?